src/elf.rs

```diff
@@ -498,6 +498,31 @@ impl<E: UserDefinedError, I: InstructionMeter> Executable<E, I> {
         })
     }
 
+    /// Calculate the total memory size of the executable
+    pub fn mem_size(&self) -> u64 {
+        let total = std::mem::size_of_val(&self)
+            // elf bytres
+            + self.elf_bytes.mem_size()
+            // ro section
+            + self.ro_section.len()
+            // text section info
+            + self.text_section_info.name.len()
+            // bpf functions
+            + self.bpf_functions.iter().fold(0, |state, (_, (val, name))| state + std::mem::size_of_val(&val) + name.len())
```

Comment:
`+ std::mem::size_of_val(&name)`?

Reply:
No, I mean you have to add that as well so that it becomes:
`state + std::mem::size_of_val(&val) + std::mem::size_of_val(&name) + name.len()`
or:
`state + std::mem::size_of::<u32>() + std::mem::size_of::<String>() + name.len()`